### ISSUE TITLE
Corrupted btree entries should not result in panics

### ISSUE THREAD

**Issue by arkpar** [paritytech/parity-db#191]
When parsing value table entries all expected reads should be validated agains actual buffer size. I believe regular hash value table already have checks in place. Btree entry parsing however does not and should be fixed.

Example of such panic: https://github.com/paritytech/parity-db/issues/190

**Comment by cheme** [comment]
Right, will check this part (sounds like a nice monday morning task :)
